include/engine/api/route_api.hpp

```diff
@@ -822,16 +823,34 @@ class RouteAPI : public BaseAPI
                             return anno.datasource;
                         });
                 }
-                if (requested_annotations & RouteParameters::AnnotationsType::Nodes)
+                if (requested_annotations & RouteParameters::AnnotationsType::Nodes ||
+                    requested_annotations & RouteParameters::AnnotationsType::OSMNodes)
                 {
                     util::json::Array nodes;
+                    util::json::Array osm_nodes;
+
                     nodes.values.reserve(leg_geometry.node_ids.size());
                     for (const auto node_id : leg_geometry.node_ids)
                     {
-                        nodes.values.push_back(
-                            static_cast<std::uint64_t>(facade.GetOSMNodeIDOfNode(node_id)));
+                        const auto osm_node_id =
```

Comment:
In case of annotations we add this `osm_nodes` if it was explicitly requested(`annotations=osm_nodes,...`) or in case of `annotations=true`